### ISSUE TITLE
Architecture refinement

### ISSUE THREAD

**Issue by RikuVan** [RikuVan/world-of-webdevandsausages#8]
Not sure about how the current layers of the app are built. The api endpoints seem ok to me in that they are only parsing/building the correct responses/requests with lens. The dtos and doas are probably ok. But what is the separation between services and controllers, especially for database operations? Right now the services are kind of pointless, with the controllers doing everything.

Is there a better way to do this? If we use a functional style we could perhaps eliminate the services for db stuff and wrap the doa requests in "Use cases", like this: https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html

**Comment by leomelin** [comment]
In my opinion, all the logic should be moved to service layer instead of controllers - controllers could do data validation & fiddle with request object but the service layer should do all the heavy lifting and call database actions. At least it's the way I see it

**Comment by leomelin** [comment]
But then again, there's that "api" -layer that is equivalent to traditional controllers - so we are basically having one layer too much here. I think the services should be removed and all the controllers should be renamed to services. The api could be renamed to controllers

**Comment by RikuVan** [comment]
works for me

**Comment by RikuVan** [comment]
actually I would like to try using the Arrow IO monad to wrap side-effects, ie. touching of the database. This can be used with coroutines under the hood to make it happen in its own thread asynchronously. So this would happen in the service.

**Comment by leomelin** [comment]
Yeah I meant the naming but I don't care if the files are called "api" or "controller" if they do basically the same thing thus handle the request-response and call the service layer for more advanced business logic
